client/admin.go

```diff
@@ -0,0 +1,190 @@
+)
+
+const (
+	// Types of admin config to manipulate.
+
+	// Accounting expects proto.AcctConfig
+	Accounting = "acct"
+	// Permission expects proto.PermConfig
+	Permission = "perms"
+	// Quit only handles Get requests.
+	Quit = "quit"
+	// User expects proto.UserConfig
+	User = "users"
+	// Zone expects proto.ZoneConfig.
+	Zone = "zones"
+)
+
+// AdminClient issues http requests to admin endpoints.
+// TODO(marc): unify the way we handle addresses in clients.
+type AdminClient struct {
+	context    *base.Context
+	address    string
+	configType string
+}
+
+// NewAdminClient returns a new 'configType' admin client, talking to 'address'.
+func NewAdminClient(ctx *base.Context, address, configType string) AdminClient {
+	return AdminClient{ctx, address, configType}
+}
+
+// adminURI builds a base URI for the embedded 'configType'
+func (a *AdminClient) adminURI() string {
+	return fmt.Sprintf("%s://%s/_admin/%s", a.context.RequestScheme(),
+		a.address, a.configType)
+}
+
+// adminURIWithKey builds a URI for the 'configType' and key.
+func (a *AdminClient) adminURIWithKey(key string) string {
+	return fmt.Sprintf("%s/%s", a.adminURI(), url.QueryEscape(key))
+}
+
+// Get issues a GET and returns the plain-text body. It cannot take a key.
+func (a *AdminClient) Get() (string, error) {
+	req, err := http.NewRequest("GET", a.adminURI(), nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add(util.AcceptHeader, util.PlaintextContentType)
+	body, err := a.do(req)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// GetJSON issues a GET request and returns a json-encoded response.
+func (a *AdminClient) GetJSON(key string) (string, error) {
+	req, err := http.NewRequest("GET", a.adminURIWithKey(key), nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add(util.AcceptHeader, util.JSONContentType)
+	body, err := a.do(req)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// GetYAML issues a GET request and returns a yaml-encoded response.
+func (a *AdminClient) GetYAML(key string) (string, error) {
+	req, err := http.NewRequest("GET", a.adminURIWithKey(key), nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add(util.AcceptHeader, util.YAMLContentType)
+	body, err := a.do(req)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// SetYAML issues a POST request for the given key using the yaml-encoded body.
+func (a *AdminClient) SetYAML(key, body string) error {
+	req, err := http.NewRequest("POST", a.adminURIWithKey(key), strings.NewReader(body))
+	if err != nil {
+		return err
+	}
+	req.Header.Add(util.ContentTypeHeader, util.YAMLContentType)
+	_, err = a.do(req)
+	return err
+}
+
+// List issues a GET request to list all configs. Returns a list of keys.
+func (a *AdminClient) List() ([]string, error) {
+	req, err := http.NewRequest("GET", a.adminURI(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := a.do(req)
```

Comment:
actually, it seems I was wrong. It was returning json because that's the default, and PlainText is not a valid encoding for this.
I'll adjust the List method to have either yaml or json.